tests/helper/helper_oc.go

```diff
@@ -0,0 +1,269 @@
+package helper
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gbytes"
+	"github.com/onsi/gomega/gexec"
+)
+
+type OcRunner struct {
+	// path to oc binary
+	path string
+}
+
+// New initializes new OcRunner
+func NewOcRunner(ocPath string) OcRunner {
+	return OcRunner{
+		path: ocPath,
+	}
+}
+
+// Run oc with given arguments
+func (oc *OcRunner) Run(cmd string) *gexec.Session {
+	session := CmdRunner(cmd)
+	Eventually(session).Should(gexec.Exit(0))
+	return session
+}
+
+// CreateRandProject create new project with random name (10 letters)
+// without writing to the config file (without switching project)
+func (oc *OcRunner) CreateRandProject() string {
+	projectName := "project-" + RandString(8)
+	fmt.Fprintf(GinkgoWriter, "Creating a new project: %s\n", projectName)
+	session := CmdRunner("oc new-project " + projectName + " --skip-config-write")
+
+	Eventually(session).Should(gexec.Exit(0))
+	Eventually(session).Should(gbytes.Say("created on server"))
+	time.Sleep(2 * time.Second)
```

Comment:
why sleep :-(

Reply:
Need to find a better solution

Reply:
Sleep is necessary atm because push is trying to create the project again even after creating the component w.r.t new test flow. I would say later we can go for a better solution to avoid sleep in test code. WDYT ?  
```
Running oc with args: [new-project project-jtjonrwq --skip-config-write] 
[oc] Project "project-jtjonrwq" created on server "https://192.168.64.3:8443".
[oc] 
[oc] To switch to this project and start adding applications, use:
[oc] 
[oc]     oc project project-jtjonrwq
Running odo with args: [create nodejs --project project-jtjonrwq nfmsom --ref master --git https://github.com/openshift/nodejs-ex] 
[odo]  •  Checking component  ...
[odo]  ✓  Checking component
[odo]  •  Checking component version  ...
[odo]  ✓  Checking component version
[odo] Please use `odo push` command to create the component with source deployed
Running odo with args: [push] 
[odo]  ✓  Creating project project-jtjonrwq
[odo]  ✗  Failed creating project project-jtjonrwq
[odo]  ✗  project.project.openshift.io "project-jtjonrwq" already exists
```